OOP/InputAction.java

```diff
@@ -0,0 +1,165 @@
+import java.util.Scanner;
+
+
+public class InputAction implements Action{
+static private Train.Type type;
+	public void  perform(int input)
+	{
+		
+	if(input==1)
+	{
+		
+		type=Train.Type.PASSENGER;
+		
+		
+	}else
+	{
+		
+		type=Train.Type.GOODS;
+		
+	}
+		
+	}
+	
+	public static void InputProcess() {
+		while (true) {
+			Scanner scanner = new Scanner(System.in);
+			System.out
+					.println("welcome to Railways ticket booking center Enter your name");
+			String name = scanner.next();
+			MenuNew menuNew=new MenuNew();
+			
+			showChart(type);
+			System.out.println("Enter from to stations");
```

Comment:
ACTION: You are still using SOP. This is not what the question and I intended for. Try to understand:-
Displaying Station names and asking to enter a station name is one menu, Displaying train names and asking to enter train name is another menu....and so on....
So what I want is, the code should be like this:
     StationMenu stationNames = new StationMenu ();
     stationNames.display();
     //train menu
     TrainMenu trainMenu = new TrainMenu();
     trainMenu.display();

like this you have to do. So your inputProcess() should not contain any menu-related-SOP.
Now, above menu classes you have to prepare...with their ActionMenuItems. If you observe the reference menu code you will find how and where to put SOP and scanner.